server/server.go

```diff
@@ -592,3 +599,13 @@ func (s *Server) Shutdown() {
 	_ = s.HTTPServer.Shutdown(ctx)
 	s.GRPC.GracefulStop()
 }
+
+func newProxyProtocolListener(httpListener net.Listener, readHeaderTimeout time.Duration) net.Listener {
+	// Wraps the listener with a proxy protocol listener.
+	// NOTE: go-proxyproto supports non-PROXY, PROXY v1 and PROXY v2 protocols via the same listener.
+	// Therefore, enabling this feature does not break existing setups.
```

Comment:
I suggest you move this comment into a more visible place. Perhaps into the README.md at the root of the repository?

Reply:
I like your suggestion. Though, the `README.md` at the root does not seem to be the right place, considering how generic it is and how little information it contains. What do you think about creating a new `README.md` in the `server` directory instead?
I noticed that other directories do have their own `README`s.

Reply:
Sounds great.

Reply:
I went for adding `PROXYPROTOCOL.md`, because the information is very specific.